openmp/libomptarget/plugins-nextgen/amdgpu/src/rtl.cpp

```diff
@@ -2869,15 +2869,14 @@ struct AMDGPUPluginTy final : public GenericPluginTy {
         if (Status != HSA_STATUS_SUCCESS)
           return Status;
 
-        // TODO: This is not allowed by the standard.
-        char ISAName[Length];
-        Status = hsa_isa_get_info_alt(ISA, HSA_ISA_INFO_NAME, ISAName);
+        llvm::SmallVector<char> ISAName(Length);
```

Comment:
Do we need `Length + 1` here? I know it's not there even before.

Reply:
Okay, this is something I wasn't sure about. The documentation states that the length does not include the null terminator, but printing the Length suggested that it's not true. Given that I'm not trusting the documentation I instead wrote it in such a way that it will work regardless by trimming any null terminators at the end if they exist.